### PR TITLE
Add unit test for search/api.go

### PR DESCRIPTION
**Pull request by liuzhengbaihe**

### CODE REVIEW

**Comment by upccup** [comment]
👍

File: src/plugins/search/router_test.go [hunk 1]
```diff
@@ -79,11 +81,42 @@ func TestRegisterApiForSearch(t *testing.T) {
...
+
 func TestIndexData(t *testing.T) {
 	searchApi := &SearchApi{}
+	searchApi.Indexer = &MockCraneIndexer{}
 	searchApi.IndexData()
-	assert.Equal(t, len(searchApi.Index), 0, "should be equal")
-	assert.Empty(t, searchApi.Store, "should be equal")
-	assert.Empty(t, searchApi.PrefetchStore, "should be equal")
-	assert.Empty(t, searchApi.Indexer, "should be equal")
+	time.Sleep(time.Minute * time.Duration(SEARCH_LOAD_DATA_INTERVAL))
```
Comment (vitan) [inline src/plugins/search/router_test.go]: @yqguodataman  the indexdb will be updated per duration?

Comment (vitan) [inline src/plugins/search/router_test.go]: how about trigger the indexer update directly than time.sleep?

File: src/plugins/search/api_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func CloseServer() {
+	searchServer.Close()
+}
+
+func TestSearch(t *testing.T) {
+	SetupServer()
+	defer CloseServer()
+	time.Sleep(time.Minute * time.Duration(SEARCH_LOAD_DATA_INTERVAL))
```
Comment (vitan) [inline src/plugins/search/api_test.go]: how about trigger the indexer update directly than time.sleep?

File: src/plugins/search/router_test.go (hunk 1 above)
Comment (liuzhengbaihe) [inline src/plugins/search/router_test.go]: good idea. Let me try.

Comment (liuzhengbaihe) [inline src/plugins/search/router_test.go]: 看了一下，不能trigger the indexer, 手动trigger会和循环执行的代码冲突。

Comment (liuzhengbaihe) [inline src/plugins/search/router_test.go]: Indexdb 每隔一分钟刷一次，一个线程无限循环在跑。

**Pull request merged**